src/pytest_workflow/workflow.py

```diff
@@ -175,6 +192,8 @@ def process(self, number_of_threads: int = 1):
             thread.start()
             threads.append(thread)
         self.join()
+        for error in self._process_errors:
```

Comment:
Tiny nitpick. The loop is a bit superfluous; we will raise on the first error, any subsequent errors won't be shown to the user. 

You could do

```python
if len(self._process_errors) > 0:
   raise self._process_errors[0]  
```